action/login.js

```diff
@@ -0,0 +1,28 @@
+export default async function login(email, password) {
+  const url = 'https://bootcamp-api.codeit.kr/api/sign-in'
+  const data = {
+    email,
+    password,
+  }
+
+  const options = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(data),
+  }
+
+  try {
+    const response = await fetch(url, options)
+    const responseData = await response.json()
+    if (!!responseData.data) {
+      localStorage.setItem('accessToken', responseData.data.accessToken)
+      return { success: true }
+    }
+    return { success: false, message: '이메일 혹은 패스워드가 잘못되었습니다.' }
```

Comment:
@ugaugaugaugaugauga 

fetch를 활용한 네트워크 통신 에러핸들링 컨벤션과 클린코드 관점에서, data필드 기준으로 체크하기보단 `response.ok` 를 기준으로 체크를 하는게 좋아보여요.

```suggestion
if (!response?.ok) {
      const errorResponse = await response.json(); // { detail: string }
      console.error(errorResponse);
      return { error: errorResponse.detail }
}

return { success: true, error: null }
```

Reply:
그런방법이 있었군요..!